opentracing-agent/src/test/java/io/opentracing/contrib/agent/OpenTracingHelperTest.java

```diff
@@ -16,15 +16,44 @@
  */
 package io.opentracing.contrib.agent;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
 
+import java.lang.reflect.Field;
+
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
+import io.opentracing.NoopTracerFactory;
 import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.mock.MockSpan;
 import io.opentracing.mock.MockTracer;
+import io.opentracing.util.GlobalTracer;
 
 public class OpenTracingHelperTest {
 
+    // Approach used in opentracing-util GlobalTracerTest to reset the global tracer
+    private static void _setGlobal(Tracer tracer) {
+        try {
+            Field globalTracerField = GlobalTracer.class.getDeclaredField("tracer");
+            globalTracerField.setAccessible(true);
+            globalTracerField.set(null, tracer);
+            globalTracerField.setAccessible(false);
```

Comment:
@jpkrohling I agree with your point.

However the GlobalTracer api is (by explicit request!) a _register-once_ api.
Therefore exposing (re-)setters or any modifiers is unwanted.
I would love to learn a better way of doing this.
Please create a [PR on opentracing-java](https://github.com/opentracing/opentracing-java/pulls) for `opentracing-util/src/test/.../GlobalTracerTest`.